src/bridges/notional/NotionalBridge.sol

```diff
@@ -0,0 +1,179 @@
+        address _token,
+        address _underlyingToken,
+        uint256 _amount
+    ) public returns (uint88) {
+        uint256 underlyingDecimals = ERC20(_underlyingToken).decimals();
+        if (_token != _underlyingToken) {
+            uint256 exchangeRateCurrent = CToken(_token).exchangeRateCurrent();
+            uint256 mantissa = 10 + underlyingDecimals;
+            uint256 underlyingAmount = (_amount * exchangeRateCurrent) / (10**mantissa);
+            return uint88(underlyingAmount);
+        }
+        if (underlyingDecimals >= 8) {
+            return uint88(_amount / 10**(underlyingDecimals - 8));
+        }
+        return uint88(_amount * 10**(8 - underlyingDecimals));
+    }
+
+    /**
+     * @notice Approve the fcash contract to use the input token and mint fcash token
+     * @dev If the input token is ETH, we need to deposit it into WETH. If the token is a ctoken,
+     * we need to call mintViaAsset. Otherwise, we just need to call mintViaUnderlying.
+     * @dev Will revert if the wrapped fcash contract has not been deployed
+     * @param _inputToken The address of the input asset
+     * @param _outputToken The address of the output asset
+     * @param _amount The amount of input token
+     * @return outputAmount The amount of fcash token we actually receive
+     */
+    function _enter(
+        address _inputToken,
+        address _outputToken,
+        uint256 _amount
+    ) internal returns (uint256) {
+        IWrappedfCash fcash = IWrappedfCash(_outputToken);
+        if (_inputToken == address(0)) {
+            WETH9(WETH).deposit{value: _amount}();
+            _inputToken = WETH;
+        }
+        (IERC20 underlyingToken, ) = fcash.getUnderlyingToken();
+        if (address(underlyingToken) == ETH) {
+            underlyingToken = IERC20(WETH);
+        }
+        uint88 fcashAmount = computeUnderlyingAmount(_inputToken, address(underlyingToken), _amount);
+        if (address(underlyingToken) != _inputToken) {
+            fcash.mintViaAsset(_amount, fcashAmount, address(this), 0);
+        } else {
+            fcash.mintViaUnderlying(_amount, fcashAmount, address(this), 0);
+        }
+        return ERC20(address(fcash)).balanceOf(address(this));
+    }
+
+    /**
+     * @notice Redeem fcash to the underlying token or the asset token
+     * @dev Call redeemToAsset or redeemToUnderlying depends on the type of input token
+     * @dev Since there might be a slippage, we need to compute the number
+     * of received tokens manually
+     * @param _inputToken The address of the input asset
+     * @param _outputToken The address of the output asset
+     * @param _amount The amount of the input asset
+     * @return outputAmount The actual received amount of the output asset
+     */
+    function _exit(
+        address _inputToken,
+        address _outputToken,
+        uint256 _amount
+    ) internal returns (uint256) {
+        bool isETH = address(_outputToken) == address(0);
```

Comment:
I think it would be cleaner and less bug prone if you simply passed to these functions AztecAsset instead of the ERC20 address. Then you could check whether the asset is ETH by checking if assetType is equal to AztecAssetType.ETH.